pvlib/tests/iotools/test_pvgis.py

```diff
@@ -509,6 +510,22 @@ def test_get_pvgis_map_variables(pvgis_tmy_mapped_columns):
     assert all([c in pvgis_tmy_mapped_columns for c in actual.columns])
 
 
+@pytest.mark.remote_data
+def test_read_pvgis_horizon():
+    pvgis_data = get_pvgis_horizon(35.171051, -106.465158)
+    horizon_data = pd.read_csv(DATA_DIR / 'test_read_pvgis_horizon.csv',
+                               index_col=0)
+    horizon_data.set_index('horizon_azimuth', inplace=True)
+    assert_frame_equal(pvgis_data, horizon_data)
+
+
+@pytest.mark.remote_data
+@pytest.mark.xfail(strict=True, reason='Ensure PVGIS raises exception')
+def test_read_pvgis_horizon_invalid_coords():
+    with pytest.warns(UserWarning, match='lat: Incorrect value'):
+        _ = get_pvgis_horizon(100, 50)  # unfeasible latitude
```

Comment:
```suggestion
def test_read_pvgis_horizon_invalid_coords():
    with pytest.raises(requests.HTTPError, match='lat: Incorrect value'):
        _ = get_pvgis_horizon(100, 50)  # unfeasible latitude
```

Reply:
@bgpierc I think I mislead you with this test. Specifically, pytest.warns would be incorrect given that the function does not produce a warning, but rather raises an error. Thus, pytest.raises should be used and the test should not fail as the error is expected. Apologies for this. But I think we're there now! :)